src/cmd_line/commands/writequit.ts

```diff
@@ -0,0 +1,104 @@
+"use strict";
+
+import * as vscode from "vscode";
+import * as fs from 'fs';
+import * as path from 'path';
+import * as node from "../node";
+import * as util from "../../util";
+import * as error from "../../error";
+import {ModeHandler} from "../../mode/modeHandler";
+
+//
+// Implements :writequit
+// http://vimdoc.sourceforge.net/htmldoc/editing.html#write-quit
+//
+export interface IWriteQuitCommandArguments extends node.ICommandArgs {
+    // arguments
+    // [++opt]
+    opt? : string;
+    optValue? : string;
+    // wq! [++opt]
+    bang? : boolean;
+    // wq [++opt] {file}
+    file? : string;
+    // wq! [++opt] {file}
+    // [range]wq[!] [++opt] [file]
+    range?: node.LineRange;
+}
+
+export class WriteQuitCommand extends node.CommandBase {
+    protected _arguments : IWriteQuitCommandArguments;
+
+    constructor(args : IWriteQuitCommandArguments) {
+        super();
+        this._name = "writequit";
+        this._shortName = "wq";
+        this._arguments = args;
+    }
+
+    get arguments() : IWriteQuitCommandArguments {
+        return this._arguments;
+    }
+
+    // Writing command. Taken as a basis from the "write.ts" file.
+    execute(modeHandler : ModeHandler) : void {
```

Comment:
Aside from lines 69, 77, this is the same as what's in `write.ts`. Any way we can just call that function instead? Maybe we can change the `execute` signature to a `bool` to indicate a success and if the write access was successful, go on and do the quit side of things?